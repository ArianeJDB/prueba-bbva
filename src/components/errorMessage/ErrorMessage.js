import React from 'react';
import './errorMessage.css';

const ErrorMessage = ({ errorMsg }) => {
    return (
            <div className="error">{errorMsg}</div>
    )
}

export default ErrorMessage;