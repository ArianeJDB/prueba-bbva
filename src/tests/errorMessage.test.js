import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import ErrorMessage from '../components/errorMessage/ErrorMessage'

describe('ErrorMessage component with a div with error className', () => {
    let wrapper;
    test('should render correctly', () => {
        wrapper = shallow(<ErrorMessage />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.error').exists()).toBe(true)
    })

    test('should render correctly with errorMsg as prop', () => {
        const errorMsg = 'Email is required'
        wrapper = shallow(<ErrorMessage errorMsg={errorMsg} />);
        
        expect(wrapper).toMatchSnapshot();
    })
})