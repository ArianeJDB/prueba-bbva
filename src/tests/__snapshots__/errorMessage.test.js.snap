// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component should render correctly 1`] = `
<Fragment>
  <div
    className="error"
  />
</Fragment>
`;

exports[`ErrorMessage component should render correctly with errorMsg as prop 1`] = `
<Fragment>
  <div
    className="error"
  >
    Email is required
  </div>
</Fragment>
`;
